### PR TITLE
Auto map employees

### PR DESCRIPTION
**Pull request by Sravanksk**
Tests Cases:

Try to Map Employees based on Employee Name
Try to Map Employees based on Employee Email
Try to Map Employees based on employee_id
Can also check mapping employees manually
Can also try updating Employee Mappings 

Nothing should break in this process

### CODE REVIEW

File: src/app/xero/settings/general-mappings/general-mappings.component.html [hunk 1]
```diff
@@ -20,7 +20,7 @@
         <div class="mat-body-strong" *ngIf="generalSettings.corporate_credit_card_expenses_object">
           Bank Account
         </div>
-        <mat-form-field floatLabel="always" appearance="outline" *ngIf="generalSettings.corporate_credit_card_expenses_object" required>
+        <mat-form-field floatLabel="always" appearance="outline" required *ngIf="generalSettings.corporate_credit_card_expenses_object">
```
Comment (ashwin1111) [inline src/app/xero/settings/general-mappings/general-mappings.component.html:23]: why is the order changed? any change in behaviour?

File: src/app/xero/settings/general-mappings/general-mappings.component.html [hunk 2]
```diff
@@ -34,9 +34,9 @@
         <mat-form-field floatLabel="always" appearance="outline" required
           *ngIf="generalSettings.sync_fyle_to_xero_payments">
           <mat-select class="general-mappings--input" placeholder="Select Payment Account"
-            formControlName="bankAccounts">
-            <mat-option *ngFor="let bankAccount of bankAccounts; let i = index" [value]="bankAccounts[i].destination_id">
-              {{bankAccounts[i].value}}
```
Comment (ashwin1111) [inline src/app/xero/settings/general-mappings/general-mappings.component.html:39]: this is related to payments sync bug fix?

File: src/app/xero/settings/generic-mappings/generic-mappings.component.html [hunk 3]
```diff
@@ -12,6 +12,9 @@
       <button class="page-info--primary-action" mat-flat-button color="primary" type="button" (click)="open()">
         New {{ getTitle(sourceField) | titlecase }} Mapping
       </button>
+      <button class="secondary-button" mat-flat-button color="secondary" type="button" (click)="triggerAutoMapEmployees()" *ngIf="sourceField === 'employee'">
```
Comment (ashwin1111) [inline src/app/xero/settings/generic-mappings/generic-mappings.component.html:15]: && general settings stuff, only if they are enabled show the btn

**Review by ashwin1111 — APPROVED** [review]

**Comment by ashwin1111** [comment]
@Sravanksk please check if destination_id has to be added in post of mappings, please to take care of that too

File: src/app/xero/settings/general-mappings/general-mappings.component.html (hunk 2 above)
Comment (Sravanksk) [reply src/app/xero/settings/general-mappings/general-mappings.component.html:39]: yes

**Comment by Sravanksk** [comment]
> @Sravanksk please check if destination_id has to be added in post of mappings, please to take care of that too. Not sure if you have already taken care along with category sync.

@ashwin1111  This is already taken care 💯

**Pull request merged**